gatling-core/src/main/scala/io/gatling/core/check/ConditionalCheck.scala

```diff
@@ -0,0 +1,87 @@
+/**
+ * Copyright 2011-2016 GatlingCorp (http://gatling.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+package io.gatling.core.check
+
+import java.util.UUID
+import com.typesafe.scalalogging.StrictLogging
+import io.gatling.commons.validation._
+import io.gatling.core.session.{ Expression, Session }
+
+object ConditionalCheck {
+  
+  implicit def conditionalCheckBuilder2Check[R, C <: Check[R]](conditionalCheckBuilder: ConditionalCheckBuilder[R, C]): ConditionalCheck[R, C] = conditionalCheckBuilder.build
+
+  def checkIf[R, C <: Check[R]](condition: Expression[Boolean])(thenCheck: C)(implicit cw: ConditionalCheckWrapper[R, C]): C = cw.wrap(ConditionalCheckBuilder((r: R, s: Session) => condition(s), thenCheck))
+
+  def checkIf[R, C <: Check[R]](condition: (R, Session) => Validation[Boolean])(thenCheck: C)(implicit cw: ConditionalCheckWrapper[R, C]): C = 
+    cw.wrap(ConditionalCheckBuilder((r: R,s: Session) => condition(r, s), thenCheck))
+
+}
+
+trait ConditionalCheckWrapper[R, C <: Check[R]] {
+  def wrap(check: ConditionalCheck[R, C]): C
+}
+
+case class ConditionalCheckBuilder[R, C <: Check[R]](condition: (R, Session) => Validation[Boolean], thenCheck: C) {
+  def build = ConditionalCheck[R, C](condition, thenCheck)
+}
+
+case class ConditionalCheck[R, C <: Check[R]](condition: (R, Session) => Validation[Boolean], thenCheck: C) extends Check[R] with StrictLogging {
+
+  val checkUuid: String = UUID.randomUUID.toString
+  
+  def performNestedCheck(nestedCheck: Check[R], response: R, session: Session)(implicit cache: scala.collection.mutable.Map[Any, Any]): Validation[CheckResult] = {
+    nestedCheck.check(response, session)
+  }
+  
+  def check(response: R, session: Session)(implicit cache: scala.collection.mutable.Map[Any, Any]): Validation[CheckResult] = {
+    val validationResult = condition(response, session).flatMap { c => 

```

Comment:
Please remove intermediate assignment and directly return (plus remove useless curly braces now there's one single chain.